posthog-server/src/main/java/com/posthog/server/internal/FlagEvaluator.kt

```diff
@@ -0,0 +1,803 @@
+                else -> compareStrings(overrideValue.toString(), value.toString(), operator)
+            }
+        } else {
+            // String comparison if numeric parsing fails
+            compareStrings(overrideValue.toString(), value.toString(), operator)
+        }
+    }
+
+    private fun compareNumbers(
+        lhs: Double,
+        rhs: Double,
+        operator: String,
+    ): Boolean {
+        return when (operator) {
+            "gt" -> lhs > rhs
+            "gte" -> lhs >= rhs
+            "lt" -> lhs < rhs
+            "lte" -> lhs <= rhs
+            else -> false
+        }
+    }
+
+    private fun compareStrings(
+        lhs: String,
+        rhs: String,
+        operator: String,
+    ): Boolean {
+        return when (operator) {
+            "gt" -> lhs > rhs
+            "gte" -> lhs >= rhs
+            "lt" -> lhs < rhs
+            "lte" -> lhs <= rhs
+            else -> false
+        }
+    }
+
+    private fun compareDates(
+        overrideValue: Any?,
+        value: Any?,
+        operator: String,
+    ): Boolean {
+        val parsedDate =
+            try {
+                parseDateValue(value.toString())
+            } catch (e: Exception) {
+                throw InconclusiveMatchException("The date set on the flag is not a valid format")
+            }
+
+        val overrideDate =
+            when (overrideValue) {
+                is Date -> overrideValue.toInstant().atZone(ZoneId.systemDefault())
+                is ZonedDateTime -> overrideValue
+                is Instant -> overrideValue.atZone(ZoneId.systemDefault())
+                is String -> {
+                    try {
+                        parseOverrideDate(overrideValue)
+                    } catch (e: Exception) {
+                        throw InconclusiveMatchException("The date provided is not a valid format")
+                    }
+                }
+                else -> throw InconclusiveMatchException("The date provided must be a string or date object")
+            }
+
+        return when (operator) {
+            "is_date_before" -> overrideDate.isBefore(parsedDate)
+            "is_date_after" -> overrideDate.isAfter(parsedDate)
+            else -> false
+        }
+    }
+
+    /**
+     * Parse date value from flag definition, supporting relative dates
+     */
+    private fun parseDateValue(value: String): ZonedDateTime {
+        // Try relative date first (e.g., "-1d", "-2w", "-3m", "-1y")
+        val relativeDate = parseRelativeDate(value)
+        if (relativeDate != null) {
+            return relativeDate
+        }
+
+        // Fall back to absolute date parsing
+        return parseOverrideDate(value)
+    }
+
+    /**
+     * Parse relative date format (e.g., "-1d" or "1d" for 1 day ago). Always produces a date in the past.
+     */
+    private fun parseRelativeDate(value: String): ZonedDateTime? {
+        val regex = Regex("^-?([0-9]+)([hdwmy])$")
```

Comment:
can be a global var so we dont have to compure the regex for every `parseRelativeDate` call